### PR TITLE
Optimize CircleCI config

### PR DESCRIPTION
**Pull request by TangRufus**
Let's aim for less than 30 seconds.

See: https://circleci.com/gh/TangRufus/workflows/sage/tree/circleci

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -1,46 +1,66 @@
...
+          executor:
+            type: executor
+        executor: << parameters.executor >>
+        steps:
+          - run: php -v
+          - checkout
+          - restore_cache:
+              keys:
+                - composer-v1-{{ checksum "composer.lock" }}
+                - composer-v1-
```
Comment (swalkinshaw) [inline .circleci/config.yml]: this meant to be here?

Comment (TangRufus) [reply .circleci/config.yml]: Yes, it tells circleci to "fallback to using the latest cache if no exact match is found (See https://circleci.com/docs/2.0/caching/)". Doc: https://circleci.com/docs/2.0/language-php/#config-walkthrough

**Comment by Log1x** [comment]
Nice. Didn't know about Orbs. I tried to Google something similar but didn't turn anything up.

**Comment by TangRufus** [comment]
Discuss: The example on https://circleci.com/docs/2.0/language-php/#config-walkthrough is using:

```yaml
 - save_cache: 
          key: node-v1-{{ checksum "package.json" }}
          paths:
            - node_modules
```

However, I believe `yarn.lock` is a better choice. Any idea?

**Comment by TangRufus** [comment]
@Log1x 
here you go: 
- https://circleci.com/docs/2.0/creating-orbs/#section=configuration
- https://github.com/ItinerisLtd/circleci-orbs-wordpress/tree/master/src

**Comment by Log1x** [comment]
> Discuss: The example on https://circleci.com/docs/2.0/language-php/#config-walkthrough is using:
> 
> ```yaml
>  - save_cache: 
>           key: node-v1-{{ checksum "package.json" }}
>           paths:
>             - node_modules
> ```
> 
> However, I believe `yarn.lock` is a better choice. Any idea?

I can't imagine it matters too much since it's just using the checksum of the file to make a unique key. I had used a non-lock file so it'd force a cache bust if the lock file is forgotten in the PR- but I guess having it sit on lock might be better so the test will fail if a new package is introduced without a new lock file present (the cache won't bust and the new package won't get installed)

**Comment by TangRufus** [comment]
> Cache is immutable on write: once a cache is written for a particular key like node-cache-master, it cannot be written to again.
>
> -- https://circleci.com/docs/2.0/caching/#writing-to-the-cache-in-workflows

If we use non-lock file checksum (allow forgotten lock files), this could be a better choice:

```yaml
# Run update so that we ignore `composer.lock`
- run: composer update -n --prefer-dist 
- save_cache:
  # Alway change cache key
  key: composer-v1-{{ checksum "composer.json" }}-{{ epoch }}
  paths:
  - vendor
```

**Comment by Log1x** [comment]
> > Cache is immutable on write: once a cache is written for a particular key like node-cache-master, it cannot be written to again.
> > -- https://circleci.com/docs/2.0/caching/#writing-to-the-cache-in-workflows
> 
> If we use non-lock file checksum (allow forgotten lock files), this could be a better choice:
> 
> ```yaml
> # Run update so that we ignore `composer.lock`
> - run: composer update -n --prefer-dist 
> - save_cache:
>     # Alway change cache key
>     key: composer-v1-{{ checksum "composer.json" }}-{{ epoch }}
>     paths:
>       - vendor
> ```

Well in that case cache would always pop because `epoch` would be different every time. I think we can just stick to `checksum` on the lock files. I think it's sane to have tests fail if a proper lock file isn't pushed.

**Comment by retlehs** [comment]
thank you @TangRufus!

**Comment by Log1x** [comment]
Screenshot

**Pull request merged**